include/seqan3/argument_parser/detail/format_ctd.hpp

```diff
@@ -0,0 +1,484 @@
+                                       build_reference_name(app_name,
+                                                            argument_id).data());
+        });
+
+        // Register ITEM callback.
+        item_argument_callbacks.push_back([this,
+                                           value,
+                                           desc,
+                                           validator,
+					   argument_id] () {
+            append_ctd_item_node(build_argument_name(argument_id),
+                                 get_type_as_gkn_string(value,
+                                                        validator),
+                                 desc,
+                                 "",
+                                 "*.*",
+                                 "true",
+                                 "false",
+                                 "");
+        });
+
+        args_counter++;
+    }
+
+    /*!\brief Builds the CTD document DOM tree and prints it to standard output.
+     *
+     * \param[in] meta The meta information that are needed for building the DOM tree.
+     */
+    void parse(argument_parser_meta_data const & meta) 
+    {
+        // Validate the application name using the regex provided by the CTD XML schema.
+        if (!std::regex_match(meta.app_name, 
+                              std::regex("[A-Za-z0-9?_\\-]+"))) {
+            throw parser_design_error("CTD requires the application name to match [A-Za-z0-9_\\-]+");
+        }
+
+        // Build the CTD document line by line inside the string stream.
+        append_ctd_declaration_node();
+        append_ctd_tool_node(meta);
+        
+        // Print the CTD file on the standard output stream.
+        std::cout << ctd_stream.str();
+
+        std::exit(EXIT_SUCCESS);
+    }
+
+    // functions are not needed for command line parsing but are part of the format interface.
+    //!\cond
+    void add_section(std::string const &) {}
+    void add_subsection(std::string const &) {}
+    void add_line(std::string const &, bool) {}
+    void add_list_item(std::string const &, std::string const &) {}
+    //!\endcond
+
+private:
+
+    /*!\brief Build the identifier for the i-th positional option.
+     *
+     * \param[in] argument_numeric_id The numeric identifier of the argument being described.
+     * \return The string representing the i-th positional option identifier.
+     */
+    std::string
+    build_argument_name(unsigned argument_numeric_id)
+    {
+        return std::string{"argument-"} + std::to_string(argument_numeric_id);
+    }
+
+    /*!\brief Build the 'referenceName' attribute of the 'mapping' node in the CTD XML file.
+     *
+     * \param[in] app_name The name of the application the parser refers to.
+     * \param[in] long_id The long identifer of the option being described.
+     * \return The reference name of the option being described.
+     */
+    std::string 
+    build_reference_name(std::string const & app_name,
```

Comment:
i am not sure but some redesign was applied such that many of these functions are already implemented by the base. But i have to check